### DISCUSSION TITLE
Smuggler & HiLo

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#17903] [Q&A] [answered]
As I'm sure some of the support staff are giggling over, we have several _hundred_ databases as we were running a true microservice architecture where each MS gets its own database AND each client gets their own set of those.

Turns out, it wasn't the easiest or cheapest thing to manage.

So, I'm consolidating all the microservices for each client into one database. Smuggling the documents / subscriptions / etc is easy however, when I bring over say `Units/1-A`, there's no associated HiLo with the Units collection, in the new DB so if I create a new Unit and it picks 1-A as the ID, it will overwrite existing data.

Is there a way to smuggle HiLo as well? Or is there a way to force HiLo at a specific # for a collection after the smuggle is done?

**Comment by ayende** [comment]
Smuggler should absolutely create those documents, we aren't filtering the hilos unless you export by collection.
The actual problem if you are merging is different, because you may have `units/1-A` in multiple databases, which one is the right one?

**Reply by ItsRobbAllen** [reply]
Ironically, that's my problem. We do filter based on collection because each microservice has context-specific copies of the data needed from the other microservices. Financials has minimized copies of the Units, Companies has sales data from Financials, etc.

So I need to smuggle the data from each database that that microservice is the authority over and nothing else. This is how we know that Unit/1-A from the Units database is the correct one and the Unit/1-A from Financials is just a stub copy.

This is a one-off conversion so I'm okay with having to do a little extra work, so long as I can script it so I can convert 60 clients safely. Is there any way to include the HiLo for a particular collection manually? Even if it's "Literally copy the HiLo document from one DB to another"?

**Reply by ItsRobbAllen** [reply]
Another point, if it helps - there are no HiLo records for the nonauthorative records because those are ETLd over anyway. Technically speaking, I should be able to copy all HiLos from each DB and be fine.

**Reply by ayende** [reply, marked as answer]
The easiest would be to just query all the `from @all_docs where startsWith(id(), 'Raven/Hilo/')` And write them out

**Reply by ItsRobbAllen** [reply]
Sorry for the delay, was on vacation - this worked perfectly fine minus a clash but that's my problem ;)
